# Google Play & App Store Data Analysis: Finding in-demand Apps

In this project, we will gather, sort, and assess the data of applications available on Google Play Store and Apple App Store. Analysis will be made on gathered data. This analysis will help the App developers to align their new ideas with market demands.

It is important to note that our working dataset for the project must only include free applciations from both the platforms. Moreover, these free apps must be directed for an English speaking audience. Since the apps we are looking for are free,it is, therefore, worthwhile for the developers to come up with apps that are in demand. This will help them get ample in-app ads revenue. 

## Opening PlayStore Dataset

In [1]:
opened_file_gp = open('googleplaystore.csv', encoding = "utf8")
from csv import reader
read_file_gp = reader(opened_file_gp)
apps_data_gp = list(read_file_gp)
andriod_header = apps_data_gp[0]
andriod = list (apps_data_gp[1:])

## Opening App Store Dataset

In [2]:
opened_file_as = open('AppleStore.csv', encoding = "utf8")
from csv import reader
read_file_as = reader(opened_file_as)
apps_data_as = list(read_file_as)
apple_header = apps_data_as[0]
apple = apps_data_as[1:]

## Defining a Function for Improved Readability of Our Datasets

In [3]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') 

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

## Exploring Datasets Using `explore_data()` Function

In [4]:
print(andriod_header)
print('\n')
explore_data(andriod, 1, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 10841
Number of columns: 13


## Important Columns in Google Play Dataset

Upon investigation, andriod dataset shows that the follwoing columns can be of interest to us:

|Column||Description|
|:---:||:---|
|App||The name of andriod app|
|Category||The category of app|
|Rating||Average rating of the app|
|Reviews||Total number of reviews|
|Installs||Total number of downloads|
|Genres||Category of the app|

In [5]:
print(apple_header)
print('\n')
explore_data(apple, 1, 5, True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


['284035177', 'Pandora - Music & Radio', '130242560', 'USD', '0.0', '1126879', '3594', '4.0', '4.5', '8.4.1', '12+', 'Music', '37', '4', '1', '1']


Number of rows: 7197
Number of columns: 16


## Important Columns in App Store Dataset

Upon investigation, the App Store dataset shows that the following columns can be of interest to us:

|Column||Description|
|:---:||:---|
|track_name||The name of apple app|
|user_rating||The average rating of app|
|prime_genre||Category of the app|
|rating_count_total||Total number of ratings|    

## Data Cleaning

### Deleting an Incomplete Row in Google Play Dataset

- The row that corresponds to 10472 index position in the list `list(andriod)` variable has no/wrong ratings value as can be seen below.
- The rating column is the third column whose value can fluctuate between 0.0-5.0.

In [6]:
andriod_header, andriod[10472]

(['App',
  'Category',
  'Rating',
  'Reviews',
  'Size',
  'Installs',
  'Type',
  'Price',
  'Content Rating',
  'Genres',
  'Last Updated',
  'Current Ver',
  'Android Ver'],
 ['Life Made WI-Fi Touchscreen Photo Frame',
  '1.9',
  '19',
  '3.0M',
  '1,000+',
  'Free',
  '0',
  'Everyone',
  '',
  'February 11, 2018',
  '1.0.19',
  '4.0 and up'])

del(andriod[10472])

- I have first deleted a row that has no/incorrect rating value. 
- Subsequently, I have converted the operation into a markdown to avoid further deletion of rows that get stored in `andriod[1072]` index position (in case Run All is used further in the project).

In [8]:
andriod_header, andriod[10472]

(['App',
  'Category',
  'Rating',
  'Reviews',
  'Size',
  'Installs',
  'Type',
  'Price',
  'Content Rating',
  'Genres',
  'Last Updated',
  'Current Ver',
  'Android Ver'],
 ['osmino Wi-Fi: free WiFi',
  'TOOLS',
  '4.2',
  '134203',
  '4.1M',
  '10,000,000+',
  'Free',
  '0',
  'Everyone',
  'Tools',
  'August 7, 2018',
  '6.06.14',
  '4.4 and up'])

The above output shows that the incomplete row has been deleted successfully.

### Identifying Duplicate Rows in Play Store Dataset

In [9]:
duplicate_app = []
unique_app = [] 

for row in andriod:
    name = row[0]
    if name in unique_app:
        duplicate_app.append(name)
    else:
        unique_app.append(name)

total_unique_app = len(unique_app)
total_duplicate_app = len(duplicate_app)
print('Total number of unique apps:', total_unique_app)
print('\n')
print('Total number of duplicate apps:', total_duplicate_app)
print('\n')
print('Examples of duplicate apps:', duplicate_app[:10])

Total number of unique apps: 9659


Total number of duplicate apps: 1181


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


### Frequency Table of Duplicate Apps in Play Store Dataset

In [10]:
No_of_rep = {}
for row in duplicate_app:
    name = row
    if name in No_of_rep:
        No_of_rep [name] += 1
    else:
        No_of_rep [name] = 1
        
print(No_of_rep['8 Ball Pool'])
print('\n')
print(andriod_header)
print('\n')
for row in andriod:
    name = row[0]
    if name == '8 Ball Pool':
        print(row)
        print('\n')
        

6


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['8 Ball Pool', 'GAME', '4.5', '14198297', '52M', '100,000,000+', 'Free', '0', 'Everyone', 'Sports', 'July 31, 2018', '4.0.0', '4.0.3 and up']


['8 Ball Pool', 'GAME', '4.5', '14198602', '52M', '100,000,000+', 'Free', '0', 'Everyone', 'Sports', 'July 31, 2018', '4.0.0', '4.0.3 and up']


['8 Ball Pool', 'GAME', '4.5', '14200344', '52M', '100,000,000+', 'Free', '0', 'Everyone', 'Sports', 'July 31, 2018', '4.0.0', '4.0.3 and up']


['8 Ball Pool', 'GAME', '4.5', '14200550', '52M', '100,000,000+', 'Free', '0', 'Everyone', 'Sports', 'July 31, 2018', '4.0.0', '4.0.3 and up']


['8 Ball Pool', 'GAME', '4.5', '14201891', '52M', '100,000,000+', 'Free', '0', 'Everyone', 'Sports', 'July 31, 2018', '4.0.0', '4.0.3 and up']


['8 Ball Pool', 'GAME', '4.5', '14201604', '52M', '100,000,000+', 'Free', '0', 'Everyone', 'Sports', 'July 31, 2018',

In the above output:

-  By using `No_of_rep{}` dictionary, I have made a frequency table of the duplicate apps to show the number of times they come in the Play Store dataset.

- I have shown how a specific app '8 Ball Pool' repeats 6 times in the dataset.

- Upon deep investigation of the repeated rows above, it can be seen that the reviews column (the 4th column)) varies even if the app repeats.

- This highlights that the data is gathered at different time.

- This also provides us the ground to select the exact row for a repeated app. We will select such a repeated row for our analysis that has the maximum number of reviews and will discard all the other rows for the same app.

### Creating a Dictionary `reviews_max{}` to Store Unique Apps with Maximum Reviews

In [11]:
reviews_max = {}

for row in andriod:
    name = row[0]
    n_reviews = float(row[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    
    elif name not in reviews_max:
        reviews_max[name] = n_reviews

In [12]:
print('Total number of Unique apps with maximum reviews:', len(reviews_max))
print('\n')
print('Maximum Reviews for the app 8 Ball Pool:', reviews_max['8 Ball Pool'])
print('\n')
print('Total number of unique apps with their maximum number of reviews:', len(reviews_max))



Total number of Unique apps with maximum reviews: 9659


Maximum Reviews for the app 8 Ball Pool: 14201891.0


Total number of unique apps with their maximum number of reviews: 9659


- Number of reviews for '8 Ball Pool' shows that `reviews_max{}` dictionary is indeed storing apps with their maximum nummber of reviews.
- This is because out of all the seven entries for '8 Ball Pool', 14201891.0 is the maximum number of reviews the app has gotten. A fact that can be confirmed from the output of 'Frequency Table of Duplicate Apps in Play Store Dataset'.

### Trimming the Dataset to Unique Apps with Maximum Reviews Only 

In [13]:
andriod_clean = []
already_added = []

for row in andriod:
    name = row[0]
    reviews = float(row[3])
    
    if  reviews == reviews_max[name] and (name not in already_added):
        andriod_clean.append(row)
        already_added.append(name)

In [14]:
explore_data(andriod_clean, 0, 5, True)


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Everyone', 'Art & Design', 'March 26, 2017', '1.0', '2.3 and up']


Number of rows: 9659
Number of columns: 13


The list `andriod_clean[]` is our trimmed dataset that has:
- Unique apps.
- Maximum reviews that correspond to each app.
- No incomplete row

Our Original Play Store dataset had 10841 rows. Our new cleaned dataset (with maximum review values for each app, without repititions, and without incomplete rows) has 9659 rows.

**Note: `already_added[]` list within the for loop and if statement is helping us to eliminate such cases where one app has more than one entry with similar number of reviews**.

### Building a Function to Remove Non-English Apps

In [15]:
def english(string):
   
    for element in string:
        if ord(element) > 127:
            return False
       
    else:
        return True

print(english('Instagram'))
print(english('IGI😜'))

True
False


This function can, however, also remove those English apps that have some non-ASCII symbols to them such as emojis. Therefore, in its current shape, the function cannot be used to filter English and non-English apps. **To minimize the impact of data loss, we'll only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range.**

In [16]:
def english(string):
    non_english_character = []
    for element in string:
        if ord(element) > 127:
            non_english_character.append(element)
           
    if len(non_english_character) > 3:
                return False  
    else:
            return True

print(english ('Docs To Go™😜😜 Free Office Suite'))  
print(english ('Docs To Go™😜😜😜 Free Office Suite'))  

True
False


**Now we will apply the above created function to both our data sets i.e. Google Play Store and App Store. This will help us weed out non-English apps from both the datasets.**

In [17]:
andriod_english = []
apple_english = []

for row in andriod_clean:
    name = row[0]
    if english(name):
        andriod_english.append(row)
        
for row in apple:
    name = row[1]
    if english(name):
        apple_english.append(row)
        
        
print(andriod_header)
print('\n')
explore_data(andriod_english, 0,3, True)
print('\n')
print(apple_header)
print('\n')
explore_data(apple_english, 0,3, True)


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215'

So far in the data cleaning process, we:

- Removed inaccurate data
- Removed duplicate app entries
- Removed non-English apps

As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps; we'll need to isolate only the free apps for our analysis.

**Isolating the free apps will be our last step in the data cleaning process.**

In [18]:
final_andriod = []
final_apple = []

for row in andriod_english:
    price = row[7]
    
    if price == '0':
        final_andriod.append(row)
        
for row in apple_english:
    price = float(row[4])
    
    if price == 0.0:
        final_apple.append(row)
        
explore_data(final_andriod, 0, 4, True)
print('\n')
explore_data(final_apple, 0, 4, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


Number of rows: 8864
Number of columns: 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '

With this step, our data cleaning process is complete. We have:

- Deleted incomplete rows from our dataset
- Eliminated duplicate rows with the criteria of leaving only those rows that have maximum number of ratings for any given app.
- Filtered non-English apps
- Remvoed paid apps

## Data Analysis

As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

- Build a minimal Android version of the app, and add it to Google Play.
- If the app has a good response from users, we develop it further.
- If the app is profitable after six months, we build an iOS version of the app and add it to the App Store.

Because our end goal is to add the app on both Google Play and the App Store, we need to find app profiles that are successful on both markets. For instance, a profile that works well for both markets might be a productivity app that makes use of gamification. 

For the andriod app dataset, following columns will help us in identifying the popular apps:

- App
- Genre
- Installs
- Rating

For the app store dataset, following columns will help us in identifying the popular apps:

- track_name
- prime_genre
- rating_count_tot
- user_rating

**For the first step, we will build a frequency table for genre column and prime_genre column to see the number of apps that fall in a specific type/category.**

### The Frequency Table With Percentage

In [19]:
#First Function to display frequecy of any column

def frequency (dataset, column_index):
    
    list = []
    total = 0
    for row in dataset:
        total += 1
        value = row[column_index]
        list.append(value)
    
    freq = {}
    
    for row in list:
        value = row
        if value in freq:
            freq[value] += 1
        else:
            freq[value] = 1
    
    percentages = {}
    
    for key in freq:
        
        percentage = (freq[key] / total) * 100      
        percentages[key] = percentage
        
    
    return percentages

#Second Function to Arrange Frequency in Descending Order 

def display_table(dataset, index):
    table = frequency (dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0]) 
        


### Percentage of Apps on App Store (`prime_genre` wise)

In [20]:
display_table(final_apple, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


### Percentage of Apps on Play Store (With Respect to `Category`)

In [21]:
display_table(final_andriod, 1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

### Percentage of Apps on Play Store (With Respect to `Genres`)

In [22]:
genre_apple = display_table(final_andriod, -4)


Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

### Average number of user ratings per app genre on the App Store

In [23]:
genre_apple = frequency(final_apple, -5)

for entry in genre_apple:
    total = 0
    len_genre = 0
    
    for row in final_apple:
        genre = row[-5]
     
        if genre == entry:
            total += int(row[5])
            len_genre += 1
   
    avg_rating_genre = total/len_genre
    print(entry, ':', avg_rating_genre)
        
        
    

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


We will select every genre for which, on an average, more than 50,000 people have rated. We will see apps within each of this genre to get a better idea of what is actually making these generes more popular with public. 

In [24]:
for row in final_apple:
    if row[-5] == "Navigation":
        print(row[1], ':', row[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


In [25]:
for row in final_apple:
    if row[-5] == "Reference":
        print(row[1], ':', row[5])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Google Translate : 26786
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
Merriam-Webster Dictionary : 16849
Night Sky : 12122
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
WWDC : 762
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
VPN Express : 14
Real Bike Traffic Rider Virtual Reality Glasses : 8
教えて!goo : 0
Jishokun-Japanese English Dictionary & Translator : 0


In [26]:
for row in final_apple:
    if row[-5] == "Social Networking":
        print(row[1], ':', row[5])

Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, Meet : 39819
Hangouts : 36404
LINE PLAY - Your Avatar World : 34677
WeChat : 34584
Badoo - Meet New People, Chat, Socialize. : 34428
Followers + for Instagram - Follower Analytics : 28633
GroupMe : 28260
Marco Polo Video Walkie Talkie : 27662
Miitomo : 2

In [27]:
for row in final_apple:
    if row[-5] == "Music":
        print(row[1], ':', row[5])

Pandora - Music & Radio : 1126879
Spotify Music : 878563
Shazam - Discover music, artists, videos & lyrics : 402925
iHeartRadio – Free Music & Radio Stations : 293228
SoundCloud - Music & Audio : 135744
Magic Piano by Smule : 131695
Smule Sing! : 119316
TuneIn Radio - MLB NBA Audiobooks Podcasts Music : 110420
Amazon Music : 106235
SoundHound Song Search & Music Player : 82602
Sonos Controller : 48905
Bandsintown Concerts : 30845
Karaoke - Sing Karaoke, Unlimited Songs! : 28606
My Mixtapez Music : 26286
Sing Karaoke Songs Unlimited with StarMaker : 26227
Ringtones for iPhone & Ringtone Maker : 25403
Musi - Unlimited Music For YouTube : 25193
AutoRap by Smule : 18202
Spinrilla - Mixtapes For Free : 15053
Napster - Top Music & Radio : 14268
edjing Mix:DJ turntable to remix and scratch music : 13580
Free Music - MP3 Streamer & Playlist Manager Pro : 13443
Free Piano app by Yokee : 13016
Google Play Music : 10118
Certified Mixtapes - Hip Hop Albums & Mixtapes : 9975
TIDAL : 7398
YouTube Mu

In [28]:
for row in final_apple:
    if row[-5] == "Weather":
        print(row[1], ':', row[5])

The Weather Channel: Forecast, Radar & Alerts : 495626
The Weather Channel App for iPad – best local forecast, radar map, and storm tracking : 208648
WeatherBug - Local Weather, Radar, Maps, Alerts : 188583
MyRadar NOAA Weather Radar Forecast : 150158
AccuWeather - Weather for Life : 144214
Yahoo Weather : 112603
Weather Underground: Custom Forecast & Local Radar : 49192
NOAA Weather Radar - Weather Forecast & HD Radar : 45696
Weather Live Free - Weather Forecast & Alerts : 35702
Storm Radar : 22792
QuakeFeed Earthquake Map, Alerts, and News : 6081
Moji Weather - Free Weather Forecast : 2333
Hurricane by American Red Cross : 1158
Forecast Bar : 375
Hurricane Tracker WESH 2 Orlando, Central Florida : 203
FEMA : 128
iWeather - World weather forecast : 80
Weather - Radar - Storm with Morecast App : 78
Yurekuru Call : 53
Weather & Radar : 37
WRAL Weather Alert : 25
Météo-France : 24
JaxReady : 22
Freddy the Frogcaster's Weather Station : 14
Almanac Long-Range Weather Forecast : 12
TodayAir

### Average Number of Installs per `category` for the Google Play Data Set

In [29]:
genre = frequency(final_andriod, 1)
for entry in genre:
    installs = 0.0
    count = 0.0
    for row in final_andriod:
        app = row[1]
        if app == entry:
            a_install = row[5]
            one = a_install.replace('+', '')
            two = float(one.replace(',', ''))
            installs += two
            count += 1

    avg_installs = installs/count   
    print(entry, ':' , avg_installs)
        
        
        
        
        
        
        

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_